### DISCUSSION TITLE
NodeJS support

### DISCUSSION THREAD

**Discussion by redeemefy** [zed-industries/zed#6633] [Feature Requests]
In a brand new project with `0` dependencies.
```
src
 |-index.js
package.json
```
```json
{
  "name": "test-node-api",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "volta": {
    "node": "20.9.0",
    "npm": "10.2.3"
  }

```
```javascript
import http from 'node:...'; // ctrl+space or crtl+: doesn't trigger any intellisense
```
In vscode I could do `ctrl+space` and the suggestion from vscode shows in the same project without any dependencies. Is there anything I need to do to enable this? I'm using the wrong key binding? Or, do I need to do `npm install --save-dev @types/node`?

### UPDATE 04NOV2023
By doing `npm install --save-dev @types/node` suggestions are working. I just thought that just because I see that the language server `JavaScript` is showing in the status bar I was going to get Node intellisense out of the box without installing `@types/node`.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
